fastlane/fastlane
# [action]Adds action for obtaining remote repo default branch

### Checklist

### Motivation and Context
This introduces a new action to obtain the default branch name from a remote git repository. Doing this can support other actions such as creating merge requests by ensuring that the proper branch name is provided as a target. With the shift away from "master", it is beneficial to have this kind of flexibility.

### Description

A new action has been created in the source control category that enables retrieving a remote git repository's default branch.

It was tested under the following conditions:
- In a folder with no available git repo (an empty string was returned)
- In a folder with a git repo, but no available remote (an empty string was returned)
- In a folder with a valid git repo and the local branch on the default remote (correct branch name returned)
- In a folder with a valid git repo and the local branch NOT on the default remote (correct branch name returned)

### Testing Steps


- Scenario 1, test in a folder with no git repo. Run `bundle exec fastlane run git_default_remote_branch`
- Scenario 2, test in a folder only a local git repo.
  - Init a git repo, commit a file
  - Run `bundle exec fastlane run git_default_remote_branch` 
- Scenario 3, test in a folder with a valid remote repo
  - Clone a repo, ensure you are on the default branch
  - Run `bundle exec fastlane run git_default_remote_branch` . Default name should be seen 
- Scenario 4, test in a folder with a valid remote repo while not on default branch
  - Clone a repo, switch to a non-default branch
  - Run `bundle exec fastlane run git_default_remote_branch` . Default name should be seen
